theory/DDsmu/countpairs_s_mu_impl.c.src

```diff
@@ -250,29 +250,47 @@ int countpairs_s_mu_DOUBLE(const int64_t ND1, DOUBLE *X1, DOUBLE *Y1, DOUBLE *Z1
         }
     }
 
-    if(options->boxsize < 0 && options->periodic){
-        fprintf(stderr, "boxsize = %g must be >= 0 with periodic wrap. Please specify a non-zero boxsize or zero to detect the particle extent.\n",
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
+#endif
+
+    if(options->periodic && options->boxsize == BOXSIZE_NOTGIVEN){
+        fprintf(stderr, "boxsize = %g must be specified with periodic wrap. Please specify a non-zero boxsize, or zero to detect the particle extent, or -1 to make a dimension non-periodic.\n",
             options->boxsize);
         return EXIT_FAILURE;
     }
-    const DOUBLE xdiff = options->boxsize > 0 ? options->boxsize:(xmax-xmin);
-    const DOUBLE ydiff = options->boxsize > 0 ? options->boxsize:(ymax-ymin);
-    const DOUBLE zdiff = options->boxsize > 0 ? options->boxsize:(zmax-zmin);
-    if(options->verbose && options->periodic) {
-        fprintf(stderr,"Running with points in [xmin,xmax] = %lf,%lf with periodic wrapping = %lf\n",xmin,xmax,xdiff);
-        fprintf(stderr,"Running with points in [ymin,ymax] = %lf,%lf with periodic wrapping = %lf\n",ymin,ymax,ydiff);
-        fprintf(stderr,"Running with points in [zmin,zmax] = %lf,%lf with periodic wrapping = %lf\n",zmin,zmax,zdiff);
-    }
+    
+    const double boxsize_y = options->boxsize_y == BOXSIZE_NOTGIVEN ? options->boxsize : options->boxsize_y;
+    const double boxsize_z = options->boxsize_z == BOXSIZE_NOTGIVEN ? options->boxsize : options->boxsize_z;
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
 
-    if(get_bin_refine_scheme(options) == BINNING_DFL) {
-        if(smax < 0.05*xdiff) {
-            options->bin_refine_factors[0] = 1;
+    const int periodic_x = options->periodic && options->boxsize_x >= 0;
+    const int periodic_y = options->periodic && boxsize_y >= 0;
+    const int periodic_z = options->periodic && boxsize_z >= 0;
+
+    const DOUBLE xwrap = (options->periodic && options->boxsize_x > 0) ? options->boxsize_x : (xmax-xmin);
+    const DOUBLE ywrap = (options->periodic && boxsize_y > 0) ? boxsize_y : (ymax-ymin);
+    const DOUBLE zwrap = (options->periodic && boxsize_z > 0) ? boxsize_z : (zmax-zmin);
```

Comment:
Think I asked this before as well - shouldn't x/y/zwrap be set to 0 when the relevant boxsize_x/y/z is < 0?

Also, if we are only testing for a negative number for setting non-periodic, then the python docstrings need to be updated. Currently, the docstrings say to set to -1.0 for non-periodic along any axis

Reply:
So I thought it didn't matter, but indeed we do end up passing `xwrap` to `get_binsize_DOUBLE`, so this had the side effect of disallowing Rmax > L/2 for non-periodic boxes, when really Rmax should be unrestricted.  The expanded brute-force test caught this.  So now we do set `xwrap = 0` for non-periodic.

Reply:
We actually use `BOXSIZE_NOTGIVEN = -2.` as a special value in the C API, so the docstrings are correct that -1 is the right value to pass.

Reply:
Let me know if you want to add the comment - otherwise, I will merge this in

Reply:
I'm happy with the new comment; ready for merge, I think!